libs/beacon/include/beacon/beacon_manager.hpp

```diff
@@ -59,187 +52,125 @@ class BeaconManager
   struct SignedMessage
   {
     Signature signature;
-    PublicKey public_key;
     Identity  identity;
   };
 
-  BeaconManager(CertificatePtr cert = nullptr)
-    : certificate_{cert}
-  {
-    if (certificate_ == nullptr)
-    {
-      auto ptr = new crypto::ECDSASigner();
-      certificate_.reset(ptr);
-      ptr->GenerateKeys();
-    }
-  }
+  BeaconManager(CertificatePtr = nullptr);
 
   BeaconManager(BeaconManager const &) = delete;
   BeaconManager &operator=(BeaconManager const &) = delete;
 
-  void SetCertificate(CertificatePtr cert)
+  void SetCertificate(CertificatePtr certificate)
```

Comment:
Would you mind putting everything in a cpp that can be?

Reply:
No problem.